autopilot/graph.go

```diff
@@ -83,19 +84,33 @@ func (d dbNode) Addrs() []net.Addr {
 // NOTE: Part of the autopilot.Node interface.
 func (d dbNode) ForEachChannel(cb func(ChannelEdge) error) error {
 	return d.node.ForEachChannel(d.tx, func(tx *bolt.Tx,
-		ei *channeldb.ChannelEdgeInfo, ep, _ *channeldb.ChannelEdgePolicy) error {
+		ei *channeldb.ChannelEdgeInfo, _, _ *channeldb.ChannelEdgePolicy) error {
 
-		pubkey, _ := ep.Node.PubKey()
+		peer, err := ei.OtherNodeKeyBytes(d.node.PubKeyBytes[:])
+		if err != nil {
+			return err
+		}
+
+		peerKey, err := btcec.ParsePubKey(peer, btcec.S256())
+		if err != nil {
+			return err
+		}
+		peerNode, err := d.g.db.FetchLightningNode(peerKey)
+		if err != nil {
+			return err
+		}
+		
```

Comment:
Ok, semantics change compared to master, but I think it is better like this. As you say, policies are not required for the auto pilot.

But the change become bigger than I liked, because of the lookup of `LightningNode` which was previously done inside `LightningNode.ForEachChannel`. Needed to gain access to `db` somehow.

I kept the changes apart in commit d258efe4ecc480145a199d1213de6816e3d45e39. Maybe there is a cleaner way?

Reply:
Hm, yeah this might not even work, since calling `FetchLightningNode` inside an already open DB transaction might cause problems. 

I don't see an immediate way of achieving what we want without doing more changes to the database to make it possible to fetch the node without the edge policy. I'm okay with keeping your original implementation (skipping a nil-policy) with an added TODO saying that we ideally should support the case where the policy is nil.

Reply:
Ok, pulling out that commit.